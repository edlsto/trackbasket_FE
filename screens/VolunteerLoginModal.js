import React, { useState, useContext, useEffect } from 'react';
import { AsYouType } from 'libphonenumber-js';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';

import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  Alert,
  KeyboardAvoidingView,
} from 'react-native';
import VolunteerContext from '../volunteer-context';
import { TextField } from '../components/Form';
import { Button } from '../components/Button';
import Constants from 'expo-constants';

const VolunteerLoginModal = ({ navigation }) => {
  const {
    volunteer,
    setVolunteer,
    setInstallationId,
    installationId,
    location,
  } = useContext(VolunteerContext);
  const asYouType = new AsYouType('US');

  useEffect(() => {
    setInstallationId(Constants.deviceId);
  }, []);

  const [name, setName] = useState(handleNameValue('name'));
  const [phone, setPhone] = useState(handleNameValue('phone'));

  useEffect(() => {
    setInstallationId(Constants.deviceId);
  }, []);

  const initiateFormatter = () => {
    if (phone.length > 1) {
      return asYouType.input(phone);
    }
  };

  const handleSubmit = () => {
    if (!name) {
      Alert.alert('Please enter your name');
    } else if (!phone) {
      Alert.alert('Please enter your phone');
    } else {
      const info = {
        name,
        phone,
        installationId,
        location,
      };
      setVolunteer(info);
      navigation.navigate('VolunteerHome');
    }
  };

  function handleNameValue(type) {
    if (!volunteer) {
      return '';
    } else {
      return volunteer[type];
    }
  }

  return (
    <KeyboardAvoidingView style={styles.container} behavior="padding">
      <KeyboardAwareScrollView>
        <View style={styles.containerInner}>
          {!volunteer && (
            <Text style={styles.introtext}>
              Sign up to volunteer to shop for a person in need
            </Text>
          )}
          {volunteer && <Text style={styles.introtext}>Update your info</Text>}
          <Text style={styles.h2}> * All Fields Required</Text>

          <TextField
            label="Name"
            placeholder="Name"
            onChangeText={setName}
            value={name}
          />

          <TextField
            label="Phone number"
            placeholder="Phone number"
            onChangeText={setPhone}
            keyboardType="numeric"
            value={initiateFormatter()}
          />
          <View style={styles.btnContainer}>
            <Button onPress={handleSubmit} text="SUBMIT" />
          </View>
        </View>
      </KeyboardAwareScrollView>
    </KeyboardAvoidingView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    borderWidth: 1,
  },
  containerInner: {
    justifyContent: 'center',
    width: 300,
    marginTop: 70,
    marginBottom: 50,
  },
  introtext: {
    fontSize: 20,
    marginBottom: 30,
    textAlign: 'center',
    fontFamily: 'HelveticaNeue-Bold',
  },
  btnContainer: {
    marginTop: 10,
  },
  h2: {
    color: 'red',
    fontSize: 16,
    fontFamily: 'HelveticaNeue-BoldItalic',
    marginBottom: 20,
  },
});

export default VolunteerLoginModal;
