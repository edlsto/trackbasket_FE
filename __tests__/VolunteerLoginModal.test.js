import React from 'react';
import VolunteerLoginModal from '../screens/VolunteerLoginModal';
import renderer from 'react-test-renderer';

test('renders correctly', async () => {
  const tree = renderer.create(<VolunteerLoginModal />).toJSON();
  expect(tree).toMatchSnapshot();
});
