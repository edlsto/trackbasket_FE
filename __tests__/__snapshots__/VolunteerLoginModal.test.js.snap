// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderWidth": 1,
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "justifyContent": "center",
            "marginBottom": 50,
            "marginTop": 70,
            "width": 300,
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "HelveticaNeue-Bold",
              "fontSize": 20,
              "marginBottom": 30,
              "textAlign": "center",
            }
          }
        >
          Sign up to volunteer to shop for a person in need
        </Text>
        <Text
          style={
            Object {
              "color": "red",
              "fontFamily": "HelveticaNeue-BoldItalic",
              "fontSize": 16,
              "marginBottom": 20,
            }
          }
        >
           * All Fields Required
        </Text>
        <View>
          <Text
            style={
              Object {
                "fontSize": 14,
                "marginBottom": 3,
              }
            }
          >
            Name
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Name"
            rejectResponderTermination={true}
            style={
              Object {
                "borderColor": "black",
                "borderWidth": 1,
                "fontSize": 15,
                "height": 35,
                "marginBottom": 5,
                "padding": 5,
              }
            }
            testID="Form.TextInput"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View>
          <Text
            style={
              Object {
                "fontSize": 14,
                "marginBottom": 3,
              }
            }
          >
            Phone number
          </Text>
          <TextInput
            allowFontScaling={true}
            keyboardType="numeric"
            maxLength={14}
            onChangeText={[Function]}
            placeholder="Phone number"
            rejectResponderTermination={true}
            style={
              Object {
                "borderColor": "black",
                "borderWidth": 1,
                "fontSize": 15,
                "height": 35,
                "marginBottom": 5,
                "padding": 5,
              }
            }
            testID="Form.TextInput"
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#59DE7E",
                "opacity": 1,
                "paddingHorizontal": 40,
                "paddingVertical": 10,
                "width": 220,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontFamily": "HelveticaNeue-Bold",
                    "fontSize": 24,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              SUBMIT
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
